src/plasma_manager_tests.h

```diff
@@ -0,0 +1,113 @@
+/**
+ * This file contains all definitions that are internal to the plasma manager
```

Comment:
How about we merge it with plasma_manager.h? Is that possible? If not, I'd consider naming it plasma_manager_impl.h or something else that doesn't contain the word "test"; having "test" in the name very much implies that it is only used for testing.